x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -352,6 +356,23 @@ public Query termQuery(Object value, SearchExecutionContext context) {
             throw new IllegalArgumentException(CONTENT_TYPE + " fields do not support term query");
         }
 
+        @Override
+        public Query existsQuery(SearchExecutionContext context) {
+            if (getEmbeddingsField() == null) {
+                return new MatchNoDocsQuery();
+            }
+
+            // Do the equivalent of a nested query with an exists query for the embeddings field
+            String nestedFieldPath = getChunksFieldName(name());
+            BitSetProducer parentFilter = new QueryBitSetProducer(Queries.newNonNestedFilter(IndexVersion.current()));
```

Comment:
Yes this is annoying and error prone. I wonder if we should simplify things a bit and only allow the semantic text field to be defined at the root level? It's a very simple change to do ;). 
Currently we allow the semantic text field to be defined anywhere so we have to be careful when we automatically build the inner nested queries. We don't have tests of nested nested semantic text so I assume that it doesn't even work today.

Reply:
Agreed that we should restrict `semantic_text` fields from being declared in a nested field, at least for now. We have no test coverage in this area and I agree that it probably doesn't even work.

However, I still think we should make the above change, if only because it is simple and it's one less thing to update when we do want to support `semantic_text` fields in nested fields.

Reply:
Thanks for the heads up. I've done 3dc6c1d8dd62e8a9ee22c8bec86e423fda548bdb  to expose a toQuery() method in `NestedQueryBuilder` so we don't need to copy / paste code. WDYT @Mikep86 @jimczi ?

I will tackle not allowing `semantic_text` as a nested field as well.

Reply:
Avoid usage in nested fields in [this PR](https://github.com/elastic/elasticsearch/pull/110055)